System/RaspberryPi/GPIO.hs

```diff
@@ -164,6 +173,19 @@ foreign import ccall unsafe "bcm2835.h bcm2835_aux_spi_transfer"       c_transfe
 --Sets the SPI clock divider and therefore the SPI clock speed.
 foreign import ccall unsafe "bcm2835.h bcm2835_aux_spi_setClockDivider"     c_setClockDividerAUXSPI :: CUShort -> IO ()
 
+------------------------------------------- PWM functions --------------------------------------------------------------------------
+-- sets the PWM clock
+foreign import ccall unsafe "bcm2835.h bcm2835_pwm_set_clock" c_setClockPWM :: CUShort -> IO ()
+
+-- sets the PWM pulse ratio
+foreign import ccall unsafe "bcm2835.h bcm2835_pwm_set_data" c_setDataPWM :: CUChar -> CUShort -> IO ()
+
+-- sets the mode of the given PWM channel
+foreign import ccall unsafe "bcm2835.h bcm2835_pwm_set_mode" c_setModePWM :: CUChar -> CUChar -> CUChar -> IO ()
+
+-- sets the maximum range of the PWM output
+foreign import ccall unsafe "bcm2835.h bcm2835_pwm_set_range" c_setRangePWM :: CUChar -> CUChar -> CUChar -> IO ()

```

Comment:
This type signature does not seem to match the one of the C library function, which takes a `uint8_t` and a `uint32_t` instead of 3 `uint8_t`s: [http://www.airspayce.com/mikem/bcm2835/group__pwm.html#ga9f5ca04d4f859d96f1205a03c79de7ce](http://www.airspayce.com/mikem/bcm2835/group__pwm.html#ga9f5ca04d4f859d96f1205a03c79de7ce).